### PR TITLE
(//bazel): Native compilation support for Jetson AGX platform

### PR DESCRIPTION
**Pull request by andi4191**
# Description
It includes native compilation support for NVIDIA Jetson AGX platform.

Dependencies:
Cross-compiled libraries from PyTorch on Jetson. Refer https://ngc.nvidia.com/catalog/containers/nvidia:l4t-pytorch

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.
- Breaking change (x86_64 bazel system needs to adapt and co-exist with aarch64 support)
- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation and have regenerated the documentation (`make html` in docsrc)
- [ ] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: third_party/libs/cudnn.BUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (narendasan) [inline third_party/libs/cudnn.BUILD:1]: move into `//third_party/cudnn`

**Comment by narendasan** [comment]
Also we need to change the sphinx documentation to document the new usage for aarch64 (//docsrc/tutorials/installation.rst)

**Comment by narendasan** [comment]
Closes #36

File: tests/core/converters/converter_test.bzl [hunk 2]
```diff
@@ -9,6 +9,7 @@ def converter_test(name, visibility=None):
             "//core",
             "@googletest//:gtest_main",
         ] + select({
+            ":aarch64":  ["@libtorch_aarch64//:libtorch"],
```
Comment (narendasan) [inline tests/core/converters/converter_test.bzl:12]: I neglected to remove these

File: tests/modules/BUILD [hunk 3]
```diff
@@ -74,6 +74,7 @@ cc_library(
         "//tests/util",
         "@googletest//:gtest_main",
     ] + select({
+        ":aarch64" : ["@libtorch_aarch64//:libtorch"],
```
Comment (narendasan) [inline tests/modules/BUILD:77]: this one as well is not necessary

File: tests/util/BUILD [hunk 4]
```diff
@@ -19,18 +34,21 @@ cc_library(
...
-            "@libtorch_pre_cxx11_abi//:libtorch",
-            "@libtorch_pre_cxx11_abi//:caffe2",
-        ],
-        "//conditions:default":  [
-            "@libtorch//:libtorch",
-            "@libtorch//:caffe2",
-        ],
-    })
+    ] +select({
+        ":aarch64": ["@libtorch_aarch64//:libtorch",
```
Comment (narendasan) [inline tests/util/BUILD:41]: Same in this case, the aarch64 and pre_cxx11_abi_aarch64 arent needed

**Comment by narendasan** [comment]
@andi4191 can you rebase this on master the PR? is trying to recommit a bunch of old changes

**Comment by narendasan** [comment]
There were some issues with the PR, I filed one against your branch to resolve them. I think that we need to get this merged ASAP, it is currently blocking CUDA 11 work. I think if you rebase to remove those extra commits we should be good to merge this and then do documentation and DLA work in separate PRs.

**Comment by andi4191** [comment]
Closing this PR. Updated PR tracked here #124

**Pull request closed without merging**